src/textfont.js

```diff
@@ -226,6 +246,18 @@ export class TextFont {
     return (metric.advanceWidth / this.resolution) * this.pointsToPixels;
   }
 
+  getWidthForString(s) {
+    const key = this.textHashBase + '-' + s;
+    let width = 0;
+    if (!TextFont.fontTextHash[key]) {
+      for (let j = 0; j < s.length; ++j) {
+        width += this.getWidthForCharacter(s[j]);
+      }
+      TextFont.fontTextHash[key] = width;
```

Comment:
I think the name `textWidthCache` is more descriptive here.

Reply:
ok, I have `TextFont.textWidthCache[this.fontCacheKey] ` as the first level.